modules/contact/test/tests/explicit_dynamics/deep_impact.i

```diff
@@ -0,0 +1,267 @@
+    variable = accel_z
+    displacement = disp_z
+    first = false
+    execute_on = 'TIMESTEP_END'
+  []
+  [vel_z]
+    type = TestNewmarkTI
+    variable = vel_z
+    displacement = disp_z
+    execute_on = 'TIMESTEP_END'
+  []
+[]
+
+[Kernels]
+  [DynamicTensorMechanics]
+    displacements = 'disp_x disp_y disp_z'
+  []
+
+  [Mass_x]
+    type = MassMatrix
+    variable = disp_x
+    density = density
+    matrix_tags = 'mass'
+  []
+  [Mass_y]
+    type = MassMatrix
+    variable = disp_y
+    density = density
+    matrix_tags = 'mass'
+  []
+  [Mass_z]
+    type = MassMatrix
+    variable = disp_z
+    density = density
+    matrix_tags = 'mass'
+  []
+[]
+
+[Kernels]
+  [gravity]
+    type = Gravity
+    variable = disp_z
+    value = -981.0
+    block = 1
+  []
+[]
+
+[BCs]
+  [x_front]
+    type = DirectDirichletBC
+    variable = disp_x
+    boundary = 'ball_front'
+    value = 0.0
+  []
+  [y_front]
+    type = DirectDirichletBC
+    variable = disp_y
+    boundary = 'ball_front'
+    value = 0.0
+  []
+  [x_fixed]
+    type = DirectDirichletBC
+    variable = disp_x
+    boundary = 'base_back'
+    value = 0.0
+  []
+  [y_fixed]
+    type = DirectDirichletBC
+    variable = disp_y
+    boundary = 'base_back'
+    value = 0.0
+  []
+  [z_fixed]
+    type = DirectDirichletBC
+    variable = disp_z
+    boundary = 'base_back'
+    value = 0.0
+  []
+
+[]
+
+[ExplicitDynamicsContact]
+  [my_contact]
+    model = frictionless_balance
+    primary = 'base_front ball_back'
+    secondary = 'ball_back base_front'
+    vel_x = 'vel_x'
+    vel_y = 'vel_y'
+    vel_z = 'vel_z'
+  []
+[]
+
+[Materials]
+  [elasticity_tensor_block_one]
+    type = ComputeIsotropicElasticityTensor
+    youngs_modulus = 1e1
+    poissons_ratio = 0.3
+    block = 1
+    constant_on = SUBDOMAIN
+  []
+  [elasticity_tensor_block_two]
+    type = ComputeIsotropicElasticityTensor
+    youngs_modulus = 1e7
+    poissons_ratio = 0.3
+    block = 2
+    constant_on = SUBDOMAIN
+  []
+  [strain_block]
+    type = ComputeFiniteStrain
+    displacements = 'disp_x disp_y disp_z'
+    implicit = false
+  []
+  [stress_block]
+    type = ComputeFiniteStrainElasticStress
+  []
+  [density_one]
+    type = GenericConstantMaterial
+    prop_names = density
+    prop_values = 1e7
+    output_properties = 'density'
+    block = '1'
+  []
+  [density_two]
+    type = GenericConstantMaterial
+    prop_names = density
+    prop_values = 1e3
+    output_properties = 'density'
+    block = '2'
+  []
+  [wave_speed]
+    type = WaveSpeed
+    outputs = 'exodus'
+    output_properties = 'wave_speed'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  start_time = 0
+  end_time = 0.04
+  dt = 0.0001
+  timestep_tolerance = 1e-6
+  [TimeIntegrator]
+    type = DirectCentralDifference
+    mass_matrix_tag = 'mass'
+    use_constant_mass = true
+  []
+  skip_exception_check = true
+[]
+[Outputs]
+  interval = 100
+  exodus = true
+  checkpoint = true # for regression testing purposes
```

Comment:
where is it used by the test harness?

Reply:
I copied this from the a similar input file in the folder. In the other input file it was added in this commit: https://github.com/idaholab/moose/commit/51d4fc6c2329e0252ce1daea76c76a9df9acc224
Not sure why it is needed.